### PR TITLE
doc: adding link to nrf54l pdk in zephyr

### PR DESCRIPTION
**Pull request by mia-ko**
Updating entries for nrf54l in sample_board_rows.txt file, adding link to pdk board docs in zephyr

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Comment by mia-ko** [comment]
> Shouldn't the boards table also be updated here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/config_and_build/board_support/board_names.html#boards-included-in-sdk-zephyr ?

I was just updating the sample tables, but yes, probably. I could do the update on that page as well :D

File: doc/nrf/config_and_build/board_support/board_names.rst [hunk 1]
```diff
@@ -73,6 +73,16 @@ Also see the :ref:`zephyr:boards` section in the Zephyr documentation.
 +-------------------+------------+-------------------------------------------------------------------+---------------------------------------+
 | nRF5340 Audio     | PCA10121   | :ref:`nrf5340_audio_dk_nrf5340 <zephyr:nrf5340_audio_dk_nrf5340>` |  ``nrf5340_audio_dk_nrf5340_cpuapp``  |
 +-------------------+------------+-------------------------------------------------------------------+---------------------------------------+
+| nRF54H20 DK       | PCA10145   | :ref:`nrf54h20pdk_nrf54h20 <zephyr:nrf54h20pdk_nrf54h20>`         | ``nrf54h20pdk_nrf54h20_cpuapp``       |
```
Comment (mia-ko) [inline doc/nrf/config_and_build/board_support/board_names.rst:76]: Not sure about the placement in the table, is there a specific order the boards should be in?

Comment (divipillai) [reply doc/nrf/config_and_build/board_support/board_names.rst:76]: I am also not sure about the order. Maybe @FrancescoSer or @annwoj would have more idea on this.

**Comment by divipillai** [comment]
> > Shouldn't the boards table also be updated here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/config_and_build/board_support/board_names.html#boards-included-in-sdk-zephyr ?
> 
> I was just updating the sample tables, but yes, probably. I could do the update on that page as well :D

Thanks Mia :).

File: doc/nrf/config_and_build/board_support/board_names.rst (hunk 1 above)
Comment (annwoj) [reply doc/nrf/config_and_build/board_support/board_names.rst:76]: Haven't discussed this really, but from the perspective of this entire table I'd say the placement makes sense. @FrancescoSer do you have some additional information regarding this?

Comment (FrancescoSer) [reply doc/nrf/config_and_build/board_support/board_names.rst:76]: As Anna suggested, it appears to be organized by series, and likely within each series, it is ordered by release date. I have no strong preference, we could leave the order as is, unless you have any specific suggestion on this.

Comment (greg-fer) [reply doc/nrf/config_and_build/board_support/board_names.rst:76]: We "always" listed the series with the lowest numbers at the bottom (highest number comes first). Examples: Product Guide at the bottom of this page: https://www.nordicsemi.com/Products and in the NCS, the list of Working with guides, where nRf91 Series comes at the top.

I would suggest following this "highest number first" order, just for the sake of consistency in all of our docsets. See for example the table in https://docs.nordicsemi.com/bundle/nrf-connect-programmer/page/index.html

Comment (divipillai) [reply doc/nrf/config_and_build/board_support/board_names.rst:76]: As Greg suggested, I also agree to follow the above approach for consistency. I had searched for similar example in the other docset, and working with guides for nRF54h and nRF54l follow the above approach.

Comment (annwoj) [reply doc/nrf/config_and_build/board_support/board_names.rst:76]: Wasn't aware of this, sounds good!

**Review by mia-ko — COMMENTED** [review]
Awaiting feedback on PCA number for nrf54l15 PDK, and build targets to be documented (as there is a mismatch between build targets in the sample_board_rows.txt file and the PDK documentation in Zephyr docs).

**Review by mia-ko — COMMENTED** [review]
Dependencies to two PRs:
- https://github.com/nrfconnect/sdk-nrf/pull/14529, changing nRF54H20 PDK to DK
- https://github.com/nrfconnect/sdk-nrf/pull/14463, which will make the build target ``nrf54l15pdk_nrf54l15_cpuapp_ns`` applicable

**Review by greg-fer — APPROVED** [review]

**Comment by mia-ko** [comment]
> Awaiting feedback on PCA number for nrf54l15 PDK, and build targets to be documented (as there is a mismatch between build targets in the sample_board_rows.txt file and the PDK documentation in Zephyr docs).

PCA number for nrf54l15 PDK is correct
Build targets are correct, but the build target nrf54l15pdk_nrf54l15_cpuapp_ns won't be applicable until the PR #14463 is merged.

**Review by divipillai — APPROVED** [review]

**Review by annwoj — APPROVED** [review]

**Comment by mia-ko** [comment]
All dependencies merged in, this PR is ready to go.

**Pull request merged**